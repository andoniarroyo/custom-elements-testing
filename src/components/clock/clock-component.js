const templateString = `
    <style>
        .clock {
          font-size: 250px;
        }
    </style>
    <div class="clock">
      <span id="hours" class="hours"></span>
      <span id="minutes" class="minutes"></span>
      <span id="seconds" class="seconds"></span>
    </div>
`;
const template = document.createElement('template');
template.innerHTML = templateString;

export class ClockComponent extends HTMLElement {
  static tag = "clock-component";

  #shadowRoot = this.attachShadow({ mode: "open" }); // document;

  connectedCallback() {
     this.appendTemplate();
  }

  appendTemplate() {
    let templateContent = template.content.cloneNode(true);
    this.shadowRoot.append(templateContent);
    this.connectClock();
  }

  connectClock() {
    setInterval(() => {
      const newDate = new Date();
      this.#shadowRoot.querySelector("#hours").innerText = newDate.getHours();
      this.#shadowRoot.querySelector("#minutes").innerText = newDate.getMinutes();
      this.#shadowRoot.querySelector("#seconds").innerText = newDate.getSeconds();
    }, 1000);
  }

}

customElements.define(ClockComponent.tag, ClockComponent);