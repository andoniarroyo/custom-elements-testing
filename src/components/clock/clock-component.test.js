import "./clock-component";

const CLOCK_COMPONENT_TAG = "clock-component";

describe("The clock component", () => {

    beforeEach(() => {
        givenAClockComponent();
    });

    describe('on creating', () => {
        it('creates the expected DOM structure', () => {
            expect(thenTheClockComponent().outerHTML).toMatchSnapshot();
        });
    });

    // given
    function givenAClockComponent() {
        // clean up
        document.body.innerHTML = '';

        // creates the component with the new greeting value
        const element = document.createElement(CLOCK_COMPONENT_TAG);
        // appends it to the DOM
        document.body.appendChild(element);
    };

    // then
    function thenTheClockComponent() {
        return getClockComponent();
    };

    // helper
    function getClockComponent() {
        return document.querySelector(CLOCK_COMPONENT_TAG);
    };

});