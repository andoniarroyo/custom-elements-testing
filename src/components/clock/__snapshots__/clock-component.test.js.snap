// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The clock component on creating creates the expected DOM structure 1`] = `"<clock-component></clock-component>"`;
